src/main/java/teammates/ui/controller/AdminInstructorAccountAddAction.java

```diff
@@ -273,7 +273,13 @@ private String generateDemoCourseId(String instructorEmail) {
     */
     private String getDemoCourseIdRoot(String instructorEmail) {
         final String[] splitedEmail = instructorEmail.split("@");
-        final String head = splitedEmail[0];
+        
+        // To ensure a valid course ID is generated, replace all characters
+        // in the email address that are considered illegal in a course ID with _.
+        // Required as valid email addresses can contain special characters (such as +)
+        // that are considered illegal in a course ID
+        final String head = splitedEmail[0].replaceAll(FieldValidator.REGEX_COURSE_ID_ILLEGAL, "_");
```

Comment:
I feel safer when these small snippets are abstracted into a utility method somewhere e.g. StringHelper and thoroughly unit tested.